DirectProgramming/DPC++FPGA/Tutorials/Tools/experimental/platform-designer/Readme.md

```diff
@@ -0,0 +1,252 @@
+
+Follow these steps to compile and test the design:
+1. Compile oneAPI code to RTL using a Windows or Linux machine. 
+
+   Linux:
+
+   ```bash
+   mkdir build
+   cd build
+   cmake ..
+   make fpga_ip_export
+   ```
+
+   Windows:
+
+   ```bash
+   mkdir build
+   cd build
+   cmake -G "NMake Makefiles" ..
+   nmake fpga_ip_export
+   ```
+
+   For more details, see the Readme in the `add-oneapi` directory.
+
+2. Copy the oneAPI-generated IP to the Quartus project. Since this design uses host pipes and the `-fsycl-device-code-split=per_kernel` flag, you will need to hunt for the `.prj_X` directory that contains the IP you want to use in your design. 
+
+   You can identify the correct `.prj_X` folder by looking for the one that contains a `*_di_inst.v` whose interfaces match your kernel. For example, in this project, `add_xample.fpga_ip.prj_1` and `add_xample.fpga_ip.prj_5` are the correct `.prj_x` directories, because `add_example_fpga_ip_1_di_inst.v` and `add_example_fpga_ip_5_di_inst.v` both contain only a CSR Agent interface in addition to the clock/reset signals:
+   
+   ```
+   add_example_fpga_ip_1_di add_example_fpga_ip_1_di_inst (
+     // Interface: clock (clock end)
+     .clock                          ( ), // 1-bit clk input
+     // Interface: clock2x (clock end)
+     .clock2x                        ( ), // 1-bit clk input
+     // Interface: resetn (conduit end)
+     .resetn                         ( ), // 1-bit data input
+     // Interface: device_exception_bus (conduit end)
+     .device_exception_bus           ( ), // 64-bit data output
+     // Interface: kernel_irqs (interrupt end)
+     .kernel_irqs                    ( ), // 1-bit irq output
+     // Interface: csr_ring_root_avs (avalon end)
+     .csr_ring_root_avs_read         ( ), // 1-bit read input
+     .csr_ring_root_avs_readdata     ( ), // 64-bit readdata output
+     .csr_ring_root_avs_readdatavalid( ), // 1-bit readdatavalid output
+     .csr_ring_root_avs_write        ( ), // 1-bit write input
+     .csr_ring_root_avs_writedata    ( ), // 64-bit writedata input
+     .csr_ring_root_avs_address      ( ), // 5-bit address input
+     .csr_ring_root_avs_byteenable   ( ), // 8-bit byteenable input
+     .csr_ring_root_avs_waitrequest  ( )  // 1-bit waitrequest output
+   );
+   ```
+
+   Linux:
+
+   ```
+   cp -r add-oneapi/build/add.fpga_ip_export.prj_1/ add-quartus/
+   cp -r add-oneapi/build/add.fpga_ip_export.prj_5/ add-quartus/
+   ```
+
+   Windows:
+
+   ```
+   xcopy add-oneapi\build\add.fpga_ip_export.prj_1\ add-quartus\add.fpga_ip_export.prj_1 /e /s /i
+   xcopy add-oneapi\build\add.fpga_ip_export.prj_5\ add-quartus\add.fpga_ip_export.prj_5 /e /s /i
+   ```
+
+3. Correct the generated `_hw.tcl` file by running the `*_di_hw_tcl_adjustment_script.py` script in the generated .prj directory.
```

Comment:
It doesn't really matter where this happens; I run the 'cleanup' script after copying because in general I don't trust scripts like this not to make a mess of something else :) (We tested it but... you never know.)